cmd/frontend/main.go

```diff
@@ -159,16 +159,14 @@ func main() {
 	}
 }
 
-func forward(logger log.Logger, target *url.URL, transport http.RoundTripper) http.Handler {
-	client := http.Client{Transport: transport}
-
+func authenticate(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		username := os.Getenv(authUsernameEnv)
 		password := os.Getenv(authPasswordEnv)
 		if len(username) > 0 && len(password) > 0 {
-			level.Info(logger).Log("msg", "AUTH_USERNAME and AUTH_PASSWORD are set, handling request with basic auth")
 			reqUser, reqPass, ok := req.BasicAuth()
 			if !ok {
+				w.Header().Set("WWW-Authenticate", `Basic realm="restricted", charset="UTF-8"`)
```

Comment:
Curious, why this is needed in your case?

Reply:
I see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate, but do you have any system that actually uses this other than good practice? (:

Reply:
My point is that realm restricted is not very useful, is it? 🤔